test/libsolidity/util/TestFileParser.cpp

```diff
@@ -0,0 +1,267 @@
+#include <stdexcept>
+
+using namespace dev;
+using namespace langutil;
+using namespace solidity;
+using namespace dev::solidity::test;
+using namespace std;
+
+namespace
+{
+	void expect(std::string::iterator& _it, std::string::iterator _end, std::string::value_type _c)
+	{
+		if (_it == _end || *_it != _c)
+			throw std::runtime_error(std::string("Invalid test expectation. Expected: \"") + _c + "\".");
+		++_it;
+	}
+
+	template<typename IteratorType>
+	void skipWhitespace(IteratorType& _it, IteratorType _end)
+	{
+		while (_it != _end && isspace(*_it))
+			++_it;
+	}
+
+	template<typename IteratorType>
+	void skipSlashes(IteratorType& _it, IteratorType _end)
+	{
+		while (_it != _end && *_it == '/')
+			++_it;
+	}
+}
+
+string TestFileParser::bytesToString(bytes const& _bytes, TestFileParser::ByteFormat const& _format)
+{
+	// TODO: Convert from compact big endian if padded
+	bytes byteRange{_bytes};
+	stringstream resultStream;
+	switch(_format.type)
+	{
+		case TestFileParser::ByteFormat::SignedDec:
+			if (*_bytes.begin() & 0x80)
+			{
+				for (auto& v: byteRange)
+					v ^= 0xFF;
+				resultStream << "-" << fromBigEndian<u256>(byteRange) + 1;
+			}
+			else
+				resultStream << fromBigEndian<u256>(byteRange);
+			break;
+		case TestFileParser::ByteFormat::UnsignedDec:
+			resultStream << fromBigEndian<u256>(byteRange);
+			break;
+	}
+	return resultStream.str();
+}
+
+pair<bytes, TestFileParser::ByteFormat> TestFileParser::stringToBytes(std::string _string)
+{
+	bytes result;
+	ByteFormat format;
+	auto it = _string.begin();
+	while (it != _string.end())
+	{
+		if (isdigit(*it) || (*it == '-' && (it + 1) != _string.end() && isdigit(*(it + 1))))
+		{
+			format.type = TestFileParser::ByteFormat::UnsignedDec;
+
+			bool isNegative = (*it == '-');
+			if (isNegative)
+				format.type = TestFileParser::ByteFormat::SignedDec;
+
+			auto valueBegin = it;
+			while (it != _string.end() && !isspace(*it) && *it != ',')
+				++it;
+
+			bytes newBytes;
+			u256 numberValue(std::string(valueBegin, it));
+			// TODO: Convert to compact big endian if padded
+			if (numberValue == u256(0))
+				newBytes = bytes{0};
+			else
+				newBytes = toBigEndian(numberValue);
+			result += newBytes;
+		}
+		else
+			BOOST_THROW_EXCEPTION(std::runtime_error("Test expectations contain invalidly formatted data."));
+
+		skipWhitespace(it, _string.end());
+		if (it != _string.end())
+			expect(it, _string.end(), ',');
+		skipWhitespace(it, _string.end());
+	}
+	return make_pair(result, format);
+}
+
+vector<TestFileParser::FunctionCall> TestFileParser::parseFunctionCalls()
+{
+	vector<TestFileParser::FunctionCall> calls;
+	while (advanceLine())
+	{
+		if (m_scanner.eol())
+			continue;
+
+		TestFileParser::FunctionCall call;
+		call.signature = parseFunctionCallSignature();
+		call.value = parseFunctionCallValue();
+		call.arguments = parseFunctionCallArgument();
+
+		if (!advanceLine())
+			throw runtime_error("Invalid test expectation. No result specified.");
```

Comment:
Perhaps we should define a new exception here? What does the syntax tests parser do?

Reply:
This should even be `BOOST_THROW_EXCEPTION(runtime_error(...))`. And that's what the syntax test also does. They are then being handled by `isoltest` and `boostTest`, respectively. I think introducing a new exception is a good idea.